features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/metric/networksettled/NetworkSettledMetricResolverTest.kt

```diff
@@ -298,6 +325,45 @@ internal class NetworkSettledMetricResolverTest {
         assertThat(metricValue).isNull()
     }
 
+    @Test
+    fun `M return null W resolveMetric(){ resource was dropped }`(forge: Forge) {
+        // Given
+        val startTimestamp = System.nanoTime()
+        val fakeResourceContext = forge.getForgery<InternalResourceContext>()
+        testedMetric.resourceWasStarted(InternalResourceContext(fakeResourceContext.resourceId, startTimestamp))
+        testedMetric.resourceWasDropped(fakeResourceContext.resourceId)
+
+        // When
+        val metricValue = testedMetric.resolveMetric()
+
+        // Then
+        assertThat(metricValue).isNull()
+    }
+
+    // endregion
+
+    // region view stopped
+
+    @Test
+    fun `M cleanup W viewWasStopped`(forge: Forge) {
+        // Given
+        val startTimestamps = forge.forgeStartTimestamps()
+        val fakeResourcesIds = forge.aList(size = startTimestamps.size) {
+            forge.getForgery<UUID>().toString()
+        }
+        fakeResourcesIds.forEachIndexed { index, id ->
+            testedMetric.resourceWasStarted(
+                InternalResourceContext(id, startTimestamps[index])
+            )
+        }
+
```

Comment:
is it better to assert that `testedMetric.getResourceStartedCacheSize()` is equal to the size of the list before the view is stopped to avoid false positive of next assertion? or is there already another test case covers it?

Reply:
not sure here...when the view is stopped all the resources should be cleaned as we no longer need them in computation. The default value of that cache should always be `0` when cleared.

Reply:
I mean **before** view is stopped , what's the expected value of `getResourceStartedCacheSize`? is it also zero?

Reply:
nope...that is not 0...we could assert that also...let me add it.